### ISSUE TITLE
split

### ISSUE THREAD

**Issue by dfx1822375** [timeseriesAI/tsai#473]
How to split against your own data, how to use the get_splits () function, why should I cut, I have ZeroDivisionError when I run the program, and my program is different from the tutorial only splits.
  The following are screenshots of the program and the errors reported：
<img width="364" alt="image" src="https://user-images.githubusercontent.com/79962395/162912220-9d9d4a10-a0af-448f-9581-7ef190e85026.png">
![4 V6MT5 9QANCM~}~VFV258](https://user-images.githubusercontent.com/79962395/162913074-81a4594c-47cc-453b-99fd-f45f364f40ec.png)

**Comment by oguiza** [comment]
Hi @dfx1822375, have you tried creating the splits you need and passing them to the get_ts_dls function?

**Comment by dfx1822375** [comment]
> Hi @dfx1822375, have you tried creating the splits you need and passing them to the get_ts_dls function?

Hi @oguiza ：
Yes, my problem is that I do not know how to use the get_splits function, this function is to slice the label,？n_splits:int, valid_size, test_size, train_size these parameters in the end what is the meaning of it? How do I set them? I read the description of the function and do not really understand, what is the return of this function, and how do I operate after getting the splits? Is this function to split the training set and the test set?

**Comment by dfx1822375** [comment]
> ```python
> get_splits(y, valid_size=.2, stratify=True, random_state=123)
> ```

Hi @oguiza :
  I have figured out the use of splits, but I have a new problem: the 'NoneType' object has no attribute 'update', and I don't know why this is happening.
  Here is a screenshot of the error：
![image](https://user-images.githubusercontent.com/79962395/163508813-c9bfb194-69a2-4f94-88e3-b9e914a21780.png)

![image](https://user-images.githubusercontent.com/79962395/163508761-b07eb01d-dbbb-4677-a483-a58089f5d77a.png)

**Comment by oguiza** [comment]
Weel it seems that for some reason the chart is not displayed in your environment (I've seen that happening in Databricks for sample). You can just remove the ShowGraph callback from your Learner (although you won't be able to see the loss chart).
Please, do not submit issues with images. Copy the entire stack trace and put in between ``` .

**Comment by dfx1822375** [comment]
> Weel it seems that for some reason the chart is not displayed in your environment (I've seen that happening in Databricks for sample). You can just remove the ShowGraph callback from your Learner (although you won't be able to see the loss chart). Please, do not submit issues with images. Copy the entire stack trace and put in between ``` .

yes，I tried to remove this parameter function before and it worked fine, so is there any other way I can see the loss change in this case?
Thanks for the heads up, I'll copy the code next time

**Comment by oguiza** [comment]
> I tried to remove this parameter function before and it worked fine, so is there any other way I can see the loss change in this case? 
You can try fastai's original callback called ShowGraphCallback, but it might not work either.

**Comment by dfx1822375** [comment]
Hi @oguiza :
  How can I use  ShowGraphCallback,error reported as undefined. And when I read "10_Time_Series_Classification_and_Regression_with_MiniRocket.ipynb" , I dont know what is 'skm.accuracy_score' 
‘’‘’

PATH = Path('./models/MRL.pkl')
learn = load_learner(PATH, cpu=False)

probas, _, preds = learn.get_X_preds(new_feat)
preds
array(['6', '6', '90', ..., '42', '90', '95'], dtype='<U2')
skm.accuracy_score(y[splits[1]], preds)’‘’

And how do I test the test set after training the model?  fastai.data.transforms.Category how to convert to Numpy？
My training accuracy is about 80, why the test set accuracy is very low？
Code is here：
‘’‘dsets = TSDatasets(X_feat_new, Y_training, tfms=tfms, splits=splits)
dls   = TSDataLoaders.from_dsets(dsets.train, dsets.valid, bs=bs, batch_tfms=TSStandardize(by_var=True))
model = TST(dls.vars, dls.c, dls.len,dropout=0.3,fc_dropout=0.9)  
learn = Learner(dls, model, loss_func=LabelSmoothingCrossEntropyFlat(), metrics=[RocAuc(),accuracy])
# learn.lr_find()
print(learn.lr_find())
learn.fit_one_cycle(35, 0.0015964782268274575)  # 0.0001
probas, _, preds = learn.get_X_preds(X_test_feat_new)
print(type(preds))
print('test_label',test_label)
print('predictions',preds)’‘’
